### ISSUE TITLE
TCP/UDP library bindings?

### ISSUE THREAD

**Issue by PacketSender642** [Spydr06/CSpydr#5]
it would be fun!

**Comment by Spydr06** [comment]
Which library are you thinking of? All the basic networking functions are already implemented via linux syscalls.

**Comment by PacketSender642** [comment]
maybe this? https://github.com/starwing/znet

**Comment by Spydr06** [comment]
That would be a way to do it. The only problem is that this library is not installed on linux systems by default (maybe precompile?).  I'd rather use the builtin library from glibc/musl, but they use structs with bitfields, which CSpydr doesn't support.
Finally there would be the option to just write it from scratch, which I'll probably do someday, but currently the stdlib is too minimal for something like that...

**Comment by PacketSender642** [comment]
i think i might attempt to use that library in cspydr (i assume i d [link("the compiled library here")] right? )

**Comment by Spydr06** [comment]
yes, exactly. But that could lead to errors when the path is not correct, try using absolute paths or install the libraray in /usr/lib.
I'm really happy to see someone use my language btw! :)

**Comment by Spydr06** [comment]
So, how‘s it going? Any news?

**Comment by Spydr06** [comment]
Closing this because of lack of activity

**Comment by PacketSender642** [comment]
the "binding" stuff is in the namespace internal, and the stuff the developer is gonna use is outside it

**Comment by Spydr06** [comment]
That's good, that's also how I do it since CSpydr lacks private functions

**Comment by PacketSender642** [comment]
ah, yeah; the test will be updated soon!

also, i plan on either making a minecraft server (using the tcp stuff) or something else
